packages/cloudfront-signer/src/sign.ts

```diff
@@ -363,8 +363,13 @@ class CloudfrontSignBuilder {
     if (!date) {
       return undefined;
     }
-    const parsedDate = Date.parse(date);
-    return isNaN(parsedDate) ? undefined : this.epochTime(new Date(parsedDate));
+    let parsedDate: Date;
+    if (/^\d+$/.test(date)) {
```

Comment:
no regexes are needed. Once you update the input signature, add test cases for number and Date inputs, and pass the user value directly to the Date constructor.